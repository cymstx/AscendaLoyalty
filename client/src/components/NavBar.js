import React from "react";
import {Link} from "react-router-dom";

import "./styles/SPA.css";


function NavBar() {
	return (
			<div className="w3-bar w3-blue w3-large">
			
				<Link to="/companies" className="w3-bar-item w3-button w3-mobile">
					Our Companies
				</Link>
				<Link to="/about" className="w3-bar-item w3-button w3-mobile">
					About
				</Link>
				<Link to="/contact" className="w3-bar-item w3-button w3-mobile">Contact</Link>
				<Link to="/" className="w3-bar-item w3-button w3-hover-blue w3-center w3-mobile">
					<b>Bank of Singapore</b>
				</Link>
			</div>

	);
}
export default NavBar;
