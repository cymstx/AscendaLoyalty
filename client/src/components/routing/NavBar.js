import React, { useContext } from "react";
import "./styles/SPA.css";
import { useNavigate } from "react-router-dom"

function NavBar() {
	const navigate = useNavigate();

	const logout = ()=> {
		localStorage.clear();
		navigate('/')
	}
	return (
		<div className="w3-bar w3-blue w3-large">
			<button onClick={()=>navigate("/companies")} className="w3-bar-item w3-button w3-mobile">
				Our Companies
			</button>

			<button onClick={()=>navigate("/about")} className="w3-bar-item w3-button w3-mobile">
				About
			</button>

			<button onClick={()=>navigate("/contact")} className="w3-bar-item w3-button w3-mobile">
				Contact
			</button>
			<button onClick={()=>navigate("/home")} className="w3-bar-item w3-button w3-mobile">
				Bank of Singapore
			</button>

			<button onClick={logout} className="w3-bar-item w3-button w3-right w3-mobile">
				Logout
			</button>
		</div>
	);
}
export default NavBar;
