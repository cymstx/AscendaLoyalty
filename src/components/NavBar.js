import React from "react";
import "./SPA.css";

function NavBar() {
  return (
    <div className="w3-bar w3-blue w3-large">
      <a href="#rooms" className="w3-bar-item w3-button w3-mobile">
        Our Companies
      </a>
      <a href="#about" className="w3-bar-item w3-button w3-mobile">
        About
      </a>
      <a href="#contact" className="w3-bar-item w3-button w3-mobile">
        Contact
      </a>
      <a href="#contact" className="w3-bar-item w3-button w3-right w3-mobile">
        Submit a claim
      </a>
    </div>
  );
}
export default NavBar;
