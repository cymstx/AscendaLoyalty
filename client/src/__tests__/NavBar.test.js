import React from 'react';
import '../matchMedia.js';
import NavBar from '../components/routing/NavBar';
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom';

describe("snapshot testing", () => {
    test("snapshot for Home component", () => {
        const renderedComponent = renderer.create(
        <Router>
            <NavBar />
        </Router>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
}) 