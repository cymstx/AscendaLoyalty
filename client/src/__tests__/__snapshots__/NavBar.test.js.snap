// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Home component 1`] = `
<div
  onLoad={[Function]}
>
  <head>
    <meta
      charset="UTF-8"
    />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <link
      href="https://fonts.googleapis.com"
      rel="preconnect"
    />
    <link
      crossorigin={true}
      href="https://fonts.gstatic.com"
      rel="preconnect"
    />
    <link
      href="https://fonts.googleapis.com/css2?family=Barlow+Condensed:wght@400;700&family=Bellefair&family=Barlow:wght@400;700&display=swap"
      rel="stylesheet"
    />
  </head>
  <body>
    <nav
      className="navbar bg-dark w3-black"
    >
      <ul
        className="nav-links"
      >
        <input
          id="checkbox_toggle"
          type="checkbox"
        />
        <label
          className="hamburger"
          for="checkbox_toggle"
        >
          â˜°
        </label>
        <div
          className="menu"
        >
          <img
            src="dbs.jpg"
          />
          <li>
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="BoS_button"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="about_button"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li>
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="enquire_button"
              onClick={[Function]}
            >
              Enquire Status
            </a>
          </li>
          <li>
            <a
              class="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="resetpts"
              onClick={[Function]}
            >
              Spend money
            </a>
          </li>
          <li
            style={
              Object {
                "display": "none",
              }
            }
          >
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="create_accrual_button"
              onClick={[Function]}
            >
              Create Accrual File
            </a>
          </li>
          <li
            style={
              Object {
                "display": "none",
              }
            }
          >
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="create_handback_button"
              onClick={[Function]}
            >
              Create Handback File
            </a>
          </li>
          <li
            style={
              Object {
                "display": "none",
              }
            }
          >
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="validation"
              onClick={[Function]}
            >
              Insert Loyalty Program
            </a>
          </li>
          <li
            style={
              Object {
                "display": "none",
              }
            }
          >
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="validation"
              onClick={[Function]}
            >
              Send Daily Emails
            </a>
          </li>
          <li>
            <a
              className="ff-sans-cond uppercase text-white letter-spacing-2 fs-300"
              id="logout_button"
              onClick={[Function]}
            >
              Logout
            </a>
          </li>
        </div>
      </ul>
    </nav>
  </body>
</div>
`;
